Kitodo-API/src/main/java/org/kitodo/api/dataeditor/rulesetmanagement/MetadataViewWithValuesInterface.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.api.dataeditor.rulesetmanagement;
+
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Return value of the function
+ * {@code ComplexMetadataViewInterface.getSortedVisibleMetadatas()}. The
+ * function returns a sorted list consisting of objects of this type. It always
+ * contains the key and the list of values if values have already been defined
+ * for this key. If the key is a multiple choice, the values are grouped, hence
+ * the list, otherwise the length of the list is always only one. Effectively,
+ * the return type of
+ * {@link ComplexMetadataViewInterface#getSortedVisibleMetadatas(Collection, Collection)}
+ * is {@code List<Pair<MetadataViewInterface, Collection<T>>>}. Since the key
+ * must be repeatable if there is more than one value but the key is not a
+ * multiple choice, a map would be an inappropriate choice at this point.
+ *
+ * @param <T>
+ *            type of data objects
+ */
+public interface MetadataViewWithValuesInterface<T> {
+    /**
+     * Returns the key to be displayed at this point. The interface can return
+     * entries where the key is {@code null} with values if there were values
+     * for excluded keys.
+     *
+     * @return the key
+     */
+    Optional<MetadataViewInterface> getMetadata();
```

Comment:
Why do you use optional here?

Reply:
Because only then a meta-dataₖₑᵧ is returned, if it should also be displayed. If there is a value whose meta-dataₖₑᵧ is not to be displayed, the value is returned without a meta-dataₖₑᵧ. The value is then returned anyway, because in the changed view (after the user edited the meta-dataₑₙₜᵣᵧ) it still has to be saved again, even though it is hidden.